spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/GettableDataToMappedTypeConverter.scala

```diff
@@ -0,0 +1,254 @@
+    Cassandra user-defined-types, which could be nested. Therefore, for types that can nest other types
+    i.e. collections, options or UDTs, we make sure to use this method for any needed nested converters.
+    E.g. for a list type, we recursively call this method to get the converter for the
+    list items and then we call `TypeConverter.forType` to get a proper converter for lists.
+    */
+    val tpe = typeTag[U].tpe
+    (columnType, tpe) match {
+      case (argColumnType, TypeRef(_, Symbols.OptionSymbol, List(argScalaType))) =>
+        val argConverter = converter(argColumnType, argScalaType)
+        TypeConverter.forType[U](Seq(argConverter))
+      case (udt @ UserDefinedType(_, columns), _) =>
+        implicit val cm: ColumnMapper[U] = columnMapper[U]
+        new GettableDataToMappedTypeConverter[U](udt, udt.columnRefs)
+      case (ListType(argColumnType), TypeRef(_, _, List(argScalaType))) =>
+        val argConverter = converter(argColumnType, argScalaType)
+        TypeConverter.forType[U](Seq(argConverter))
+      case (SetType(argColumnType), TypeRef(_, _, List(argScalaType))) =>
+        val argConverter = converter(argColumnType, argScalaType)
+        TypeConverter.forType[U](Seq(argConverter))
+      case (MapType(keyColumnType, valueColumnType), TypeRef(_, _, List(keyScalaType, valueScalaType))) =>
+        val keyConverter = converter(keyColumnType, keyScalaType)
+        val valueConverter = converter(valueColumnType, valueScalaType)
+        TypeConverter.forType[U](Seq(keyConverter, valueConverter))
+      case (_, _) =>
+        TypeConverter.forType[U]
+    }
+  }
+
+  /** Returns a converter that can convert
+    * a Cassandra value of given `ColumnType` to Scala type given by `Type` */
+  private def converter(columnType: ColumnType[_], tpe: Type): TypeConverter[_] = {
+    type U = u forSome { type u }
+    implicit val tt = ReflectionUtil.typeToTypeTag[U](tpe)
+    converter[U](columnType)
+  }
+
+  /** Reads the column name of the column pointed by given column reference. */
+  private def columnName(columnRef: ColumnRef): String = {
+    columnRef.columnName
+  }
+
+  /** Returns the type of the column, basing on the struct definition. */
+  private def columnType(columnRef: ColumnRef): ColumnType[_] = {
+    columnRef match {
+      case ColumnName(columnName, _) =>
+        structDef.columnByName(columnName).columnType
+      case TTL(_, _) | WriteTime(_, _) | RowCountRef =>
+        BigIntType
+    }
+  }
+
+  /** Reads the value of the given column from the input data, without applying any conversions. */
+  private def columnValue(columnRef: ColumnRef, struct: GettableData): AnyRef = {
+    struct.getRaw(columnRef.cqlValueName)
+  }
+
+  /** Tries to convert the value with the given converter and handles the error, if any */
+  private def tryConvert(value: AnyRef, converter: TypeConverter[_], name: String): AnyRef = {
+    try
```

Comment:
Safer? Why would it be any safer for expressions? If you add another line in the middle and forget braces, the expression wouldn't compile. I think it is only desired for imperative style, when you use if without else, so this could lead to a following bug:

```scala 
if (condition)
  doSth;
  doSthElse;   // this is not a part of if
someOtherCode
```
Here, it is an expression, not a statement, and adding anything in the middle with no braces would trigger a compile error.